apiserver/facades/client/application/application_test.go

```diff
@@ -1317,21 +1280,29 @@ func (s *applicationSuite) TestSpecializeStoreOnDeployApplicationSetCharmAndAddC
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(results.Results, gc.HasLen, 1)
 	c.Assert(results.Results[0].Error, gc.IsNil)
-	c.Assert(repo.testMode, jc.IsTrue)
+	//c.Assert(repo.testMode, jc.IsTrue)
```

Comment:
remove me
but this does test a key aspect of what this test was written for, which was to ensure that the test mode was set on the repo if so configured in the model
the patched repo should record the test mode and expose it for tests to check